pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -0,0 +1,144 @@
+)
+
+const PostStartHookName = "apiserver/bootstrap-system-flowcontrol-configuration"
+
+type RESTStorageProvider struct{}
+
+var _ genericapiserver.PostStartHookProvider = RESTStorageProvider{}
+
+func (p RESTStorageProvider) NewRESTStorage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) (genericapiserver.APIGroupInfo, bool) {
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(flowcontrol.GroupName, legacyscheme.Scheme, legacyscheme.ParameterCodec, legacyscheme.Codecs)
+
+	if apiResourceConfigSource.VersionEnabled(flowcontrolv1alpha1.SchemeGroupVersion) {
+		apiGroupInfo.VersionedResourcesStorageMap[flowcontrolv1alpha1.SchemeGroupVersion.Version] = p.v1alpha1Storage(apiResourceConfigSource, restOptionsGetter)
+	}
+	return apiGroupInfo, true
+}
+
+func (p RESTStorageProvider) v1alpha1Storage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) map[string]rest.Storage {
+	storage := map[string]rest.Storage{}
+
+	// flow-schema
+	flowSchemaStorage, flowSchemaStatusStorage := flowschemastore.NewREST(restOptionsGetter)
+	storage["flowschemas"] = flowSchemaStorage
+	storage["flowschemas/status"] = flowSchemaStatusStorage
+
+	// priority-level-configuration
+	priorityLevelConfigurationStorage, priorityLevelConfigurationStatusStorage := prioritylevelconfigurationstore.NewREST(restOptionsGetter)
+	storage["prioritylevelconfigurations"] = priorityLevelConfigurationStorage
+	storage["prioritylevelconfigurations/status"] = priorityLevelConfigurationStatusStorage
+
+	return storage
+}
+
+func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
+	systemPreset := SystemPresetData{
+		FlowSchemas:                 bootstrap.SystemFlowSchemas(),
+		PriorityLevelConfigurations: bootstrap.SystemPriorityLevelConfigurations(),
+	}
+	// TODO: default flow-schemas and priority levels
+	return PostStartHookName, systemPreset.EnsureSystemPresetConfiguration(), nil
+}
+
+type SystemPresetData struct {
+	FlowSchemas                 []*flowcontrolv1alpha1.FlowSchema
+	PriorityLevelConfigurations []*flowcontrolv1alpha1.PriorityLevelConfiguration
+}
+
+func (d SystemPresetData) EnsureSystemPresetConfiguration() genericapiserver.PostStartHookFunc {
+	return func(hookContext genericapiserver.PostStartHookContext) error {
+		// Adding system priority classes is important. If they fail to add, many critical system
+		// components may fail and cluster may break.
+		err := wait.Poll(1*time.Second, 30*time.Second, func() (done bool, err error) {
+			flowcontrolClientSet, err := flowcontrolclient.NewForConfig(hookContext.LoopbackClientConfig)
+			if err != nil {
+				utilruntime.HandleError(fmt.Errorf("unable to initialize client: %v", err))
+				return false, nil
+			}
+
+			for _, flowSchema := range d.FlowSchemas {
+				_, err := flowcontrolClientSet.FlowSchemas().Get(flowSchema.Name, metav1.GetOptions{})
```

Comment:
Is it important to do a live apiserver query rather than query an informer cache at this point?

Reply:
we better to do live-query here i suppose. or it will be even more complex, we have to hold until the cache is loaded/refreshed before a query-from-cache.

Reply:
I misunderstood the `wait.Poll`.  Since this is only a start-time thing, I am less concerned about doing the live apiserver queries.